extensions/arc/deployment/src/test/java/io/quarkus/arc/test/deployment/TestArchiveMatching.java

```diff
@@ -22,6 +22,7 @@ public void testMatch() {
         ArtifactKey key = GACT.fromString(GROUP_ID + ":" + ARTIFACT_ID);
         ArtifactKey keyWithClassifier = GACT.fromString(GROUP_ID + ":" + ARTIFACT_ID + ":" + CLASSIFIER);
 
+        assertFalse(archiveMatches(key, GROUP_ID + ".different", Optional.empty(), Optional.empty()));
```

Comment:
I would also add a testcase for the positive case:

`assertTrue(archiveMatches(key, GROUP_ID, Optional.empty(), Optional.empty()));`

Reply:
Sure, good point!

Reply:
Done